### PR TITLE
Cleanup models.User.HashPassword

### PR DESCRIPTION
**Pull request by bkcsoft**
Makes it deterministic even if ran twice.

Something like this would previously break the users password:
```
u.Passwd = "foobar"
u.HashPassword()
u.HashPassword()
```

### CODE REVIEW

File: models/user.go [hunk 1]
```diff
@@ -711,7 +714,8 @@ func CreateUser(u *User) (err error) {
 	if u.Salt, err = GetUserSalt(); err != nil {
 		return err
 	}
-	u.HashPassword()
+	tempPassword := u.Passwd
+	u.HashPassword(tempPassword)
```
Comment (lafriks) [inline models/user.go]: Why variable `tempPassword` is needed?

Comment (bkcsoft) [reply models/user.go]: Probably isn't 🤔

**Comment by lunny** [comment]
LGTM except @lafriks ' comment.

**Review by thehowl — APPROVED** [review]
LGTM except for lafrik's note

**Review by appleboy — APPROVED** [review]

**Comment by lafriks** [comment]
Make LG-TM work

**Pull request merged**